src/main/java/com/combatlogger/overlay/DamageOverlay.java

```diff
@@ -0,0 +1,293 @@
+            int barX = showAvatars ? avatarSize : 0; // Bar starts after avatar if shown
+            int textX = showAvatars ? (barX + 5) : 5; // Text starts after avatar or with padding
+
+            graphics.setColor(new Color(70, 70, 70, 120));
+            graphics.fillRect(barX, yPosition, availableBarWidth, LINE_HEIGHT);
+
+            // Damage bar
+            Color semiTransparentPlayerColor = new Color(playerColor.getRed(), playerColor.getGreen(), playerColor.getBlue(), 165);
+            graphics.setColor(semiTransparentPlayerColor);
+            graphics.fillRect(barX, yPosition, barLength, LINE_HEIGHT);
+
+            // Metric
+            String secondaryText = "";
+            if (secondaryMetric == CombatLoggerConfig.SecondaryMetric.DPS) {
+                secondaryText = String.format("(%.2f, %.1f%%)", stats.getDps(), percentDamage);
+            }
+            else if (secondaryMetric == CombatLoggerConfig.SecondaryMetric.TICKS) {
+                secondaryText = String.format("(%d, %.1f%%)", stats.getTicks(), percentDamage);
+            }
+            int rowY = yPosition + ((LINE_HEIGHT - metrics.getHeight()) / 2) + metrics.getAscent();
+
+            // Damage text
+            graphics.setColor(Color.WHITE);
+            String damageText = String.format("%d %s", damage, secondaryText);
+            int damageTextXPosition = currentSize.width - metrics.stringWidth(damageText) - 2; // 2 pixels padding from the right edge
+            graphics.drawString(damageText, damageTextXPosition, rowY);
+
+            int availableNameWidth = damageTextXPosition - textX - 5; // 5 pixels padding between name and damage
+            if (availableNameWidth > 0) {
+                String truncatedNameText = truncateText(playerName, metrics, availableNameWidth);
+                graphics.drawString(truncatedNameText, textX, rowY);
+            }
+
+            yPosition += LINE_HEIGHT;
+        }
+        return new Dimension(currentSize.width, currentSize.height);
+    }
+
+    private BufferedImage loadImage(String path) {
+        BufferedImage image = null;
+        try (InputStream is = getClass().getResourceAsStream(path)) {
+            if (is != null) {
+                image = ImageIO.read(is);
+            } else {
+                log.error("Image not found at path: {}", path);
+            }
+        }
+        catch (IOException e) {
+            log.error("Error loading image at path: {}", path, e);
+        }
+        return image;
+    }
+
+    /**
+     * Truncates the given text and appends an ellipsis if it exceeds the maxWidth.
+     *
+     * @param text      The original text to potentially truncate.
+     * @param fm        The FontMetrics object for measuring text width.
+     * @param maxWidth  The maximum allowed width for the text.
+     * @return          The original or truncated text with an ellipsis.
+     */
+    private String truncateText(String text, FontMetrics fm, int maxWidth) {
```

Comment:
I did try implementing StringUtils abbreviate but it seems to take (text,maxCharacterLimit) whereas I'm working with pixels.

I played around trying to determine the amount of characters that would fill the px width we need it to, but it seemed like it would be messier than just doing the above. Especially because runescape's font isn't monospaced, would have to calculate the width of each individual character.

Reply:
it's odd to me that I couldn't find any runescape/runelite examples of truncation. Everything seems to wrap.

Reply:
> it's odd to me that I couldn't find any runescape/runelite examples of truncation. Everything seems to wrap.

Actually now that I think of it in the panel I'm pretty sure the Swing JLabel automatically does the truncation for me. Not sure if there's something we're missing with overlays that could make it work that way too